### PR TITLE
#182 Implement global tool installation for pgadmin

### LINKED ISSUE #182: Integrate PgAdmin
As a IDEasy user, I want to use PgAdmin in my PostGreSQL projects so that I can easily administrate the database.

https://www.pgadmin.org/

Can be a global tool installation.

### PR DESCRIPTION
**Pull request by slskiba**
Closes #182. 

Installation on Mac is unsuccessful due to getting a "Permission Denied" error. This problem is handled in #402.

### CODE REVIEW

File: cli/src/main/java/com/devonfw/tools/ide/tool/pgadmin/PgAdmin.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+public class PgAdmin extends GlobalToolCommandlet {
+  /**
+   * The constructor.
+   *
+   * @param context the {@link IdeContext}.
+   */
+  public PgAdmin(IdeContext context) {
+
+    //TODO: add relevant Tag.
+    super(context, "pgadmin", Set.of(Tag.MISC));
```
Comment (slskiba) [inline cli/src/main/java/com/devonfw/tools/ide/tool/pgadmin/PgAdmin.java:28]: What would be a good Tag for this tool?

Comment (hohwille) [reply cli/src/main/java/com/devonfw/tools/ide/tool/pgadmin/PgAdmin.java:28]: Seems we should create a `DB` tag for all database related tools (SQLDeveloper, PostGreSQL, Oracle XE #75).

File: cli/src/main/java/com/devonfw/tools/ide/tool/pgadmin/PgAdminUrlUpdater.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import com.devonfw.tools.ide.url.model.folder.UrlVersion;
+import com.devonfw.tools.ide.url.updater.WebsiteUrlUpdater;
+import com.devonfw.tools.ide.version.VersionIdentifier;
+
+import java.util.regex.Pattern;
+
+/**
+ * {@link WebsiteUrlUpdater} for pgadmin.
+ */
+public class PgAdminUrlUpdater extends WebsiteUrlUpdater {
```
Comment (hohwille) [inline cli/src/main/java/com/devonfw/tools/ide/tool/pgadmin/PgAdminUrlUpdater.java:13]: This is always that last resort if no other option is available.
However, I could also not find a better solution.
FYI: https://www.postgresql.org/docs/current/git.html

File: cli/src/main/resources/nls/Help.properties [hunk 3]
```diff
@@ -34,6 +34,7 @@ cmd.mvn=Tool commandlet for Maven (Build-Tool).
 cmd.node=Tool commandlet for Node.js (JavaScript runtime).
 cmd.npm=Tool commandlet for Npm (JavaScript Node Package Manager).
 cmd.oc=Tool commandlet for Openshift CLI (Kubernetes management tool).
+cmd.pgadmin=Tool commandlet for pgAdmin.
```
Comment (hohwille) [inline cli/src/main/resources/nls/Help.properties:37]: IMHO you created this feature branch from an outdated main branch.
Whenever you create a new feature branch, first do a pull of main from upstream.
With PR #389 you will be forced to also add `cmd.pgadmin.details`.

**Review by hohwille — APPROVED** [review]
@slskiba thanks for your PR. Solid work. Great that you already tested different OS and crafted a linux specific solution as well. 👍 
You will need to merge with main, resolve conflicts and then add the help details properties as well.
If review comments are addressed, this can be merged.

**Comment by slskiba** [comment]
The installWithPackageManager method of GlobalToolCommandlet has been renamed to runWithPackageManager, since it can be used to uninstall tools without changes to the code.

**Review by hohwille — APPROVED** [review]
all fine now.

**Pull request merged**